### PR TITLE
nfc: Update example

### PR DESCRIPTION
**Pull request by german77**
Updates example to compile correctly with latest libnx. Adds a bit more information about the nfc tag that is being scanned. Uses `NfpServiceType_Debug` to be able to obtain the application area id without hardcoding the value.

### CODE REVIEW

File: nfc/source/main.c [hunk 1]
```diff
@@ -194,7 +244,7 @@ int main(int argc, char* argv[])
     consoleUpdate(NULL);
 
     // Initialize the nfp:* service.
-    rc = nfpInitialize(NfpServiceType_User);
+    rc = nfpInitialize(NfpServiceType_Debug);
```
Comment (yellows8) [inline nfc/source/main.c:247]: ` // Use the NfpServiceType as required by your app, only use Debug if actually needed.`

File: nfc/source/main.c [hunk 2]
```diff
@@ -120,8 +140,40 @@ Result process_amiibo(u32 app_id) {
...
+    u32 app_id=0;
+    // Retrieve the admin info data, which contains the app id.
+    if (R_SUCCEEDED(rc)) {
+        NfpAdminInfo admin_info = {0};
+        rc = nfpGetAdminInfo(&handle, &admin_info);
+
+        if (R_SUCCEEDED(rc)) {
+            app_id = admin_info.application_area_id;
+            printf("App area: 0x%x, game ID: 0x%lx, console: ", app_id, admin_info.application_id);
+            switch(admin_info.application_area_version){
```
Comment (yellows8) [inline nfc/source/main.c:158]: Needs handling for default-case.

Comment (yellows8) [reply nfc/source/main.c:158]: `switch (admin_info.application_area_version) {`

File: nfc/source/main.c (hunk 1 above)
Comment (german77) [reply nfc/source/main.c:247]: nfpGetAdminInfo is only available on the debug interface. I could remove it if we want to keep the user interface and hardcode the application area id instead

Comment (yellows8) [reply nfc/source/main.c:247]: Can keep it, debug-usage needs a comment however.

File: nfc/source/main.c [hunk 3]
```diff
@@ -120,8 +140,43 @@ Result process_amiibo(u32 app_id) {
...
```
Comment (yellows8) [inline nfc/source/main.c:158]: `switch (admin_info.application_area_version) {`

**Review by yellows8 — APPROVED** [review]

**Pull request merged**